exporter/clickhouseexporter/exporter_metrics_test.go

```diff
@@ -0,0 +1,142 @@
+		dp.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		exemplars = dp.Exemplars().AppendEmpty()
+		exemplars.SetIntValue(54)
+		exemplars.FilteredAttributes().PutStr("key", "value")
+		exemplars.FilteredAttributes().PutStr("key2", "value2")
+
+		// histogram
+		m = sm.Metrics().AppendEmpty()
+		dpHisto := m.SetEmptyHistogram().DataPoints().AppendEmpty()
+		dpHisto.SetStartTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		dpHisto.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		dpHisto.SetCount(1)
+		dpHisto.SetSum(1)
+		dpHisto.Attributes().PutStr("key", "value")
+		dpHisto.Attributes().PutStr("key", "value")
+		dpHisto.ExplicitBounds().FromRaw([]float64{0, 0, 0, 0, 0})
+		dpHisto.BucketCounts().FromRaw([]uint64{0, 0, 0, 1, 0})
+		dpHisto.SetMin(0)
+		dpHisto.SetMax(1)
+		exemplars = dpHisto.Exemplars().AppendEmpty()
+		exemplars.SetIntValue(54)
+		exemplars.FilteredAttributes().PutStr("key", "value")
+		exemplars.FilteredAttributes().PutStr("key2", "value2")
+
+		// exp histogram
+		m = sm.Metrics().AppendEmpty()
+		dpExpHisto := m.SetEmptyExponentialHistogram().DataPoints().AppendEmpty()
+		dpExpHisto.SetStartTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		dpExpHisto.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		dpExpHisto.SetSum(1)
+		dpExpHisto.SetMin(0)
+		dpExpHisto.SetMax(1)
+		dpExpHisto.SetZeroCount(0)
+		dpExpHisto.SetCount(1)
+		dpExpHisto.Attributes().PutStr("key", "value")
+		dpExpHisto.Attributes().PutStr("key", "value")
+		dpExpHisto.Negative().SetOffset(1)
+		dpExpHisto.Negative().BucketCounts().FromRaw([]uint64{0, 0, 0, 1, 0})
+		dpExpHisto.Positive().SetOffset(1)
+		dpExpHisto.Positive().BucketCounts().FromRaw([]uint64{0, 0, 0, 1, 0})
+
+		exemplars = dpHisto.Exemplars().AppendEmpty()
+		exemplars.SetIntValue(54)
+		exemplars.FilteredAttributes().PutStr("key", "value")
+		exemplars.FilteredAttributes().PutStr("key2", "value2")
+
+		// summary
+		m = sm.Metrics().AppendEmpty()
+		summary := m.SetEmptySummary().DataPoints().AppendEmpty()
+		summary.SetStartTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		summary.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		summary.Attributes().PutStr("key", "value")
+		summary.Attributes().PutStr("key2", "value2")
+		summary.SetCount(1)
+		summary.SetSum(1)
+		quantileValues := summary.QuantileValues().AppendEmpty()
+		quantileValues.SetValue(1)
+		quantileValues.SetQuantile(1)
+	}
+	return metrics
+}
+
+func mustPushMetricsData(t *testing.T, exporter *metricsExporter, md pmetric.Metrics) {
+	err := exporter.pushMetricsData(context.TODO(), md)
+	require.NoError(t, err)
+}
+
+func newTestMetricsExporter(t *testing.T, dsn string, fns ...func(*Config)) *metricsExporter {
+	exporter, err := newMetricsExporter(zaptest.NewLogger(t), withTestExporterConfig(fns...)(dsn))
+	require.NoError(t, err)
+
+	t.Cleanup(func() { _ = exporter.Shutdown(context.TODO()) })
+	return exporter
+}
+
+func TestExporter_pushMetricsData(t *testing.T) {
```

Comment:
move this out to a _test file.

Reply:
I'm a little confused, it's already in a _test file——exporter_traces_test.go

Reply:
oh ok. Please move the test to the top of the file and move helpers down.